src/AppHost/Program.cs

```diff
@@ -43,8 +43,19 @@
 
 var blobStorage = storage.AddBlobs("eshopsupport-blobs");
 
-var pythonInference = builder.AddPythonUvicornApp("python-inference",
-    Path.Combine("..", "PythonInference"), port: 62394);
+var stage = builder.ExecutionContext.IsRunMode ? "base" : /* default stage */ null;
+var pythonInference = builder.AddDockerfile("python-inference", "../PythonInference", /* default dockerfile */ null, stage)
+    .WithHttpEndpoint(targetPort: 62394, env: "UVICORN_PORT")
```

Comment:
Previously 62394 was the `port`, not the `targetPort`. That is, you used to be able to reach the Python endpoint via requests to http://localhost:62394/classify etc. By not specifying a `port`, this becomes a random port on each startup which makes it a bit harder to verify what's going on.

I tried adding a further `port` argument here but now this error always occurs on apphost startup:

```
fail: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      could not start the proxy {"ServiceName": {"name":"python-inference"}, "Reconciliation": 25, "error": "could not start the proxy for the service: listen tcp [::1]:62395: bind: An attempt was made to access a socket in a way forbidden by its access permissions."}
```

Any ideas why this would be? Does Aspire not support setting a fixed port for Docker containers, or is some other step needed to make this work?

Reply:
Also if we change the `targetPort` to 8000, there's no need to pass `env: "UVICORN_PORT"` because it defaults to that anyway.

Reply:
Wait, the error seems specific to setting `port` values in the 62xxx range on my machine. Don't know why, but the following does work correctly for me:

 ```suggestion
    .WithHttpEndpoint(port: 57000, targetPort: 8000)
```

Are you OK with setting a fixed port like this?

Reply:
@SteveSandersonMS you're likely running into a very fun problem related to Hyper-V. On every system with Hyper-V enabled, a semi-random range of ports is reserved and can't be assigned to other applications. The exact range of ports isn't always the same and I've never come across a published set of the ports that could be reserved in this way. The end result is that, while a given fixed port may work just fine on one machine, it can often cause the issue you're running into on some other random machine. It's one reason we've generally avoided using fixed ports wherever possible in the Aspire playground samples.